code/modules/projectiles/guns/projectile.dm

```diff
@@ -225,13 +229,24 @@
 /obj/item/weapon/gun/projectile/attack_self(mob/user as mob)
 	if (target)
 		return ..()
-	if (loaded.len || stored_magazine)
+	if (loaded.len || stored_magazine || refuse.len)
 		if (load_method == SPEEDLOADER)
-			var/obj/item/ammo_casing/AC = loaded[1]
-			loaded -= AC
-			AC.forceMove(get_turf(src)) //Eject casing onto ground.
-			to_chat(user, "<span class='notice'>You unload \the [AC] from \the [src]!</span>")
-			update_icon()
+			if(!gun_flags &CHAMBER)
+				var/obj/item/ammo_casing/AC = loaded[1]
+				loaded -= AC
+				AC.forceMove(get_turf(src)) //Eject casing onto ground.
```

Comment:
this makes you able to make the casing clip through containers like lockers and mechs